sky/packages/sky/lib/src/fn3/framework.dart

```diff
@@ -712,11 +719,11 @@ abstract class BuildableElement<T extends Widget> extends Element<T> {
   /// components dirty during event handlers before the frame begins, not during
   /// the build itself.
   void markNeedsBuild() {
-    assert(!_debugStateLocked);
     assert(_debugLifecycleState == _ElementLifecycle.mounted);
     if (_dirty)
       return;
     _dirty = true;
+    assert(!_debugInBuildScope);
```

Comment:
Since you moved setting _dirty to false to after the build() function is called, I assume this is intentional. Can you elaborate on why we want markNeedsBuild()/setState() to be ok to run during that component's own build() function?

Also, document that.